modules/broker-amqp/src/main/java/org/wso2/broker/amqp/Server.java

```diff
@@ -68,7 +70,12 @@
     private Channel plainServerChannel;
     private Channel sslServerChannel;
 
-    public Server(Broker broker, AmqpServerConfiguration configuration) {
+    private boolean isHaEnabled = false;
+    private boolean isShutdownHookTriggered = false;
```

Comment:
Since this is accessed through two threads and one is on a while loop checking for its value it should be either volatile or `AtomicBoolean`.